aws-opentelemetry-distro/src/amazon/opentelemetry/distro/exporter/aws/metrics/aws_cloudwatch_emf_exporter.py

```diff
@@ -289,30 +259,165 @@ def _convert_gauge(self, metric: Metric, data_point: NumberDataPoint) -> MetricR
         record = self._create_metric_record(metric.name, metric.unit, metric.description)
 
         # Set timestamp
-        try:
-            timestamp_ms = (
-                self._normalize_timestamp(data_point.time_unix_nano)
-                if data_point.time_unix_nano is not None
-                else int(time.time() * 1000)
-            )
-        except AttributeError:
-            # data_point doesn't have time_unix_nano attribute
-            timestamp_ms = int(time.time() * 1000)
+        timestamp_ms = (
+            self._normalize_timestamp(data_point.time_unix_nano)
+            if data_point.time_unix_nano is not None
+            else int(time.time() * 1000)
+        )
         record.timestamp = timestamp_ms
 
         # Set attributes
-        try:
-            record.attributes = data_point.attributes
-        except AttributeError:
-            # data_point doesn't have attributes
-            record.attributes = {}
+        record.attributes = data_point.attributes
 
-        # For Gauge, set the value directly
-        try:
-            record.value = data_point.value
-        except AttributeError:
-            # data_point doesn't have value
-            record.value = None
+        # Set the value directly for both Gauge and Sum
+        record.value = data_point.value
+
+        return record
+
+    def _convert_histogram(self, metric: Metric, data_point: Any) -> MetricRecord:
+        """Convert a Histogram metric datapoint to a metric record.
+
+        https://github.com/mircohacker/opentelemetry-collector-contrib/blob/main/exporter/awsemfexporter/datapoint.go#L148
```

Comment:
Why are referring to a random fork and not the official OTel code?

Reply:
good catch. updated.